From 32b6c22006767f0762edfa116b8b0f7be0c5c121 Mon Sep 17 00:00:00 2001
From: Eric Le Bihan <eric.le.bihan.dev@free.fr>
Date: Wed, 27 Jul 2016 15:43:16 +0200
Subject: [PATCH] build-sys: revert use of ln relative option.

Systemd build system now uses the `--relative` option from `ln(1)`.

This option was added to GNU coreutils 8.16, which is not widely
deployed yet by GNU/Linux distributions (not available in Debian Wheezy
for example).

Signed-off-by: Eric Le Bihan <eric.le.bihan.dev@free.fr>
[Maxime: refresh the patch]
[Vincent:
  refresh the patch, move-to-rootlibdir removed by:
  https://github.com/systemd/systemd/commit/082210c7a837063fd8b520b18c221b42059d7eff
]
Signed-off-by: Maxime Hadjinlian <maxime.hadjinlian@gmail.com>
Signed-off-by: Vicente Olivert Riera <Vincent.Riera@imgtec.com>
---
 Makefile.am  | 11 ++---------
 configure.ac |  2 --
 2 files changed, 2 insertions(+), 11 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 0c27f81..4de1595 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -312,9 +312,9 @@ install-aliases-hook:
 	set -- $(SYSTEM_UNIT_ALIASES) && \
 		dir=$(systemunitdir) && $(install-aliases)
 	set -- $(USER_UNIT_ALIASES) && \
-		dir=$(userunitdir) && $(install-relative-aliases)
+		dir=$(userunitdir) && $(install-aliases)
 	set -- $(GENERAL_ALIASES) && \
-		dir= && $(install-relative-aliases)
+		dir= && $(install-aliases)

 define install-aliases
 	while [ -n "$$1" ]; do \
@@ -325,15 +325,6 @@ define install-aliases
 	done
 endef

-define install-relative-aliases
-	while [ -n "$$1" ]; do \
-		$(MKDIR_P) `dirname $(DESTDIR)$$dir/$$2` && \
-		rm -f $(DESTDIR)$$dir/$$2 && \
-		$(LN_S) --relative $(DESTDIR)$$1 $(DESTDIR)$$dir/$$2 && \
-		shift 2 || exit $$?; \
-	done
-endef
-
 install-touch-usr-hook:
 	touch -c $(DESTDIR)/$(prefix)

diff --git a/configure.ac b/configure.ac
index cf595e6..d58f303 100644
--- a/configure.ac
+++ b/configure.ac
@@ -110,8 +110,6 @@ AC_PATH_PROG([SULOGIN], [sulogin], [/usr/sbin/sulogin], [$PATH:/usr/sbin:/sbin])
 AC_PATH_PROG([MOUNT_PATH], [mount], [/usr/bin/mount], [$PATH:/usr/sbin:/sbin])
 AC_PATH_PROG([UMOUNT_PATH], [umount], [/usr/bin/umount], [$PATH:/usr/sbin:/sbin])

-AS_IF([! ln --relative --help > /dev/null 2>&1], [AC_MSG_ERROR([*** ln doesn't support --relative ***])])
-
 M4_DEFINES=

 AC_CHECK_TOOL(OBJCOPY, objcopy)
--
2.8.1

